src/services/jsTyping.ts

```diff
@@ -0,0 +1,282 @@
+// Copyright (c) Microsoft. All rights reserved. Licensed under the Apache License, Version 2.0. 
+// See LICENSE.txt in the project root for complete license information.
+
+/// <reference path='services.ts' />
+
+/* @internal */
+namespace ts.JsTyping {
+
+    interface HostType {
+        directoryExists: (path: string) => boolean;
+        fileExists: (fileName: string) => boolean;
+        readFile: (path: string, encoding?: string) => string;
+        readDirectory: (path: string, extension?: string, exclude?: string[], depth?: number) => string[];
+        writeFile: (path: string, data: string, writeByteOrderMark?: boolean) => void;
+    };
+
+    var _host: HostType;

```

Comment:
This is dangerously structured. You declare `_host` here, and use it in several top level functions in the namespace, but it's not set until the exported function `discoverTypings` is called. While those other use sites may only be called after/from `discoverTypings` currently, this may not be obvious to someone else adding or exporting more functions. Either initialize it when the namespace is initialized, or pass it from `discoverTypings` to the functions it calls that need it.